### ISSUE TITLE
Firebase function failed on PG gateway: Cannot encode value 

### ISSUE THREAD

**Issue by karthit** [cashfree/cashfree-sdk-nodejs#28]
Firebase cloud function on Cashfree PG gateway failed with error: _**Cannot encode value.**_ 

Reason: **path: undefined** (inside class PG) 

this.path variable is undefined inside Http call of PaymentGateway constructor.

**Comment by kshirish** [comment]
@karthit `Cannot encode value` seems like a Firebase error. `this.path` is supposed to be `undefined` within the constructor ... only when you use a method on an instance (pg/payouts), its value gets set.

Could you share logs? Which method are you using and how are you invoking it?

**Comment by karthit** [comment]
const pg = new PaymentGateway(pgConfig); //path: undefined
await pg.refunds.initiateRefund({req.body}); 

since the value of path: undefined, instead of null  or any default value, in Firebase cloud function this undefined value is creating this error message: _Cannot encode value_

**Comment by kshirish** [comment]
@karthit 

If you are literally doing this then it's wrong – what you are passing is an invalid object.

```js
await pg.refunds.initiateRefund({req.body});
```

Below works for me:
```js
const pgInstance = new PaymentGateway({
        env: 'TEST',
        apiVersion: '<API_VERSION>',
        appId: '<YOUR_APP_ID>',
        secretKey: '<YOUR_SECRET_KEY>',
});

const body = {
        orderId: 'ORDER123',
        referenceId: '13307',
        refundAmount: '102.00',
        refundNote: 'Sample Refund note',
        refundType: '',
        refundType: '',
        mode: '',
        accountNo: '',
        ifsc: '',
};

const response = await pgInstance.refunds.initiateRefund(body);
        console.log(response);
});
```


Also, `path` is not an issue. It's getting set as `/api/v1/order/refund` later when you call `.initiateRefund()`.

**Comment by karthit** [comment]
yes, but while working with Firebase cloud functions the PaymentGateWay constructor returns undefined path, we are getting _Cannot encode value_ error, it works with express but the issue is with only on Firebase cloud functions. 

Even though we set the path to different methods after this, Firebase cloud failed while returning this 
`const pgInstance = new PaymentGateway({
        env: 'TEST',
        apiVersion: '<API_VERSION>',
        appId: '<YOUR_APP_ID>',
        secretKey: '<YOUR_SECRET_KEY>',
});`

pgInstance object { ..., path: undefined }

**Comment by kshirish** [comment]
@karthit Why don't you try forking the repository and set a default value of `path` in the constructor as "" to see whether it works for you or not?

**Comment by kshirish** [comment]
@karthit SDK has been updated. It will work now.
